spec/workers/step_by_step_scheduled_publish_worker_spec.rb

```diff
@@ -6,25 +6,28 @@
     allow(StepNavPublisher).to receive(:publish)
   end
 
-  let(:step_by_step_page) { create(:draft_step_by_step_page, scheduled_at: Time.now) }
-
+  let(:future_datetime) { Time.now + 1.hour }
+  let!(:step_by_step_page) { create(:draft_step_by_step_page, scheduled_at: future_datetime) }
 
   it "publishes the step by step" do
-    described_class.new.perform(step_by_step_page.id)
+    Timecop.travel future_datetime do
+      described_class.new.perform(step_by_step_page.id)
 
-    expect(StepNavPublisher).to have_received(:publish).with(step_by_step_page)
+      expect(StepNavPublisher).to have_received(:publish).with(step_by_step_page)
 
-    step_by_step_page.reload
-    expect(step_by_step_page.has_been_published?).to be true
+      step_by_step_page.reload
+      expect(step_by_step_page.has_been_published?).to be true
+    end
   end
 
   it "generates a change note" do
-    described_class.new.perform(step_by_step_page.id)
-    expect(step_by_step_page.internal_change_notes.first.description).to eq("Published on schedule")
+    Timecop.travel future_datetime do
+      described_class.new.perform(step_by_step_page.id)
+      expect(step_by_step_page.internal_change_notes.first.description).to eq("Published on schedule")
+    end
   end
 
   it "doesn't publish the step by step if it isn't scheduled" do
-    step_by_step_page = create(:draft_step_by_step_page)
```

Comment:
I think this line needs to be kept in, otherwise the test doesn't make sense as there is a scheduled_at date, but it's in the future, which is what the next test is testing.

Reply:
Think this tests may need re-writing, as it passes regardless 😕

Reply:
Yes, it passes because the scheduled_at date is in the future. That's why I originally re-created the `draft_step_by_step` object without a `scheduled_at` date.

Reply:
Understood.

Reply:
Had to duplicate the creation of the step_by_step in each test otherwise got slug uniqueness validation errors 👍

Reply:
You shouldn't need to do that. I'd double check how and where the `step_by_step` object is being created.

Reply:
I needed to change it from:

```ruby
let(:step_by_step_page) { create(:draft_step_by_step_page, scheduled_at: future_datetime) }
```

...to:

```ruby
let!(:step_by_step_page) { create(:draft_step_by_step_page, scheduled_at: future_datetime) }
```

...otherwise the creation of the step by step didn't happen until we called it in the test, by which time we've used Timecop to travel to another time and we get an error saying "Scheduled at can't be in the past".

But by creating it early, it means if we then make another `create()` call in the test, then a second step_by_step is created in the same test run, and we get the slug uniqueness validation error.

At least the current version of the code is explicit about the test setup, so each test is easy to understand in isolation.

Reply:
It might be possible to get around this by moving some of the tests into a separate "context" and calling `let!` from there.